src/components/hub/elements/MessageInput.js

```diff
@@ -0,0 +1,110 @@
+import TextAreaInput from "@components/hub/elements/TextAreaInput";
+import addSuffix from "@utils/navigation/add_suffix";
+import isMobile from "@utils/get_device_type";
+import { KEY_CODES } from "@helpers/keyCodes";
+import { useEffect, useMemo, useState } from "react";
+import { useLocation } from "react-router-dom";
+
+import { ReactComponent as Attach } from "@icons/options/Attach.svg";
+import { ReactComponent as Send } from "@icons/options/Send.svg";
+
+export default function MessageInput({
+  inputTextRef,
+  onSubmitFunc,
+  isBlockedConv,
+  chatMessagesBlockRef,
+}) {
+  const location = useLocation();
+
+  const [isPrevLocationAttach, setIsPrevLocationAttach] = useState(false);
+
+  const handleInput = (e) => {
+    if (inputTextRef.current) {
+      const countOfLines = e.target.value.split("\n").length - 1;
+      inputTextRef.current.style.height = `calc(${
+        55 + countOfLines * 20 < 230 ? 55 + countOfLines * 20 : 215
+      }px * var(--base-scale)) `;
+      inputTextRef.current.scrollTop = inputTextRef.current.scrollHeight;
+    }
+  };
+
+  const handeOnKeyDown = (e) => {
+    if (
+      e.keyCode === KEY_CODES.ENTER &&
+      ((!isMobile && !e.shiftKey) || (isMobile && e.shiftKey))
+    ) {
+      e.preventDefault();
+      onSubmitFunc();
+    }
+  };
+
+  const uploadInputText = () => {
```

Comment:
probably `upload` is not the best term for this operation. Maybe `storeInputText ` ?